modules/nextflow/src/main/groovy/nextflow/ast/NextflowDSLImpl.groovy

```diff
@@ -498,182 +468,179 @@ class NextflowDSLImpl implements ASTTransformation {
 
             final args = methodCall.arguments as ArgumentListExpression
             final lastArg = args.expressions.size()>0 ? args.getExpression(args.expressions.size()-1) : null
-            final isClosure = lastArg instanceof ClosureExpression
 
-            if( isClosure ) {
-                // the block holding all the statements defined in the process (closure) definition
-                final block = (lastArg as ClosureExpression).code as BlockStatement
+            // the block holding all the statements defined in the process (closure) definition
+            final block = (lastArg as ClosureExpression).code as BlockStatement
```

Comment:
How it is inferred that this has to be a ClosureExpression type?

Reply:
Good catch, we should raise a syntax error here instead of just skipping as before